### PR TITLE
Fix for #4002: subsonicupdate plugin broken

### LINKED ISSUE #4002: Subsonicupdate plugin broken in multiple paces
### Problem

Running the latest git version of beets causes the subsonicupdate plugin to fail with a generic "Error: 1".
The current code is broken in 2 places:

1. `def __get_version(self):`
Code is trying to retrieve the version in order to decide whether to authenticate with token or password: a much better choice than my original plan of assuming everyone would use an updated version and/or prefer token over password due to additional security.
In the current code, the REST API is called by passing only parameters 'c' and 'f': this API fails because it's not complete and the plugin stops processing.

2. `def start_scan(self):`
Assuming the authentication would go through, this call would fail anyway since the current code is splitting the API version in 3 values and passing them separately, e.g. `&v=1&v=61&v=1`, rather than `&v=1.61.1`

I have always been running an ancient version of beets with my original code for the plugin and I noticed this just today as I set up a new machine and installed beets form git repo through AUR.
It also seems I am the only user of this plugin given no other feedback has been filed so far, so I can understand if this fix doesn't get high priority.

### Setup

* OS: Arch Linux
* Python version: 3.9.6
* beets version: 1.5.0
* Turning off plugins made problem go away (yes/no):  disabling subsonicupdate plugin will prevent the error from happening

### Additional Info
Subsonic API reference: http://www.subsonic.org/pages/api.jsp

### PR DESCRIPTION
**Pull request by maffo999**
## Description

Fixes #4002.
Tested locally on my Subsonic setup.

## To Do

- [X] Documentation. (If you've add a new command-line flag, for example, find the appropriate page under `docs/` to describe it.)
- [X] Changelog. (Add an entry to `docs/changelog.rst` near the top of the document.)
- [ ] Tests. (Encouraged but not strictly required.)

### CODE REVIEW

File: beetsplug/subsonicupdate.py [hunk 1]
```diff
@@ -110,48 +97,30 @@ def __format_url(endpoint):
...
-            self._log.error(u'Error: {0}'.format(error))
-            return None
-
     def start_scan(self):
         user = config['subsonic']['user'].as_str()
-        url = self.__format_url("startScan.view")
+        auth = config['subsonic']['auth'].as_str()
+        url = self.__format_url("startScan")
+        self._log.info(u'URL is {0}', url)
+        self._log.info(u'auth type is {0}', config['subsonic']['auth'])
```
Comment (sampsyo) [inline beetsplug/subsonicupdate.py:105]: These should probably be `debug` logs, as opposed to `info`? (That will make them invisible by default, and I suspect most users don't need to see them.)

File: docs/changelog.rst [hunk 2]
```diff
@@ -387,6 +387,7 @@ Fixes:
   all front images instead of blindly selecting the first one.
 * :doc:`/plugins/lyrics`: Removed the LyricWiki source (the site shut down on
   21/09/2020).
+* Fix :bug:`4002`. Subsonicupdate plugin is now functional again.
```
Comment (sampsyo) [inline docs/changelog.rst:390]: …and also has a new `auth` configuration parameter?

File: beetsplug/subsonicupdate.py [hunk 3]
```diff
@@ -34,8 +42,6 @@
 from beets.plugins import BeetsPlugin
 
 __author__ = 'https://github.com/maffo999'
-AUTH_TOKEN_VERSION = (1, 12)
-
```
Comment (sampsyo) [inline beetsplug/subsonicupdate.py:38]: I think the style checker wants this blank line preserved.

**Review by sampsyo — CHANGES_REQUESTED** [review]
Looks awesome; thanks!! Here are a couple of very minor suggestions.

**Comment by maffo999** [comment]
All should be ok now but let me know if I missed anything else.

**Review by sampsyo — APPROVED** [review]
Perfect!! Thanks so much for sorting this out! 🚀 ✨

**Pull request merged**